rpm/packagescanner.go

```diff
@@ -130,53 +130,125 @@ func (ps *Scanner) Scan(ctx context.Context, layer *claircore.Layer) ([]*clairco
 			zlog.Error(ctx).Err(err).Msg("error removing extracted files")
 		}
 	}()
+
 	// Extract tarball
 	if err := ctx.Err(); err != nil {
 		return nil, err
 	}
 	if _, err := rd.Seek(0, io.SeekStart); err != nil {
 		return nil, fmt.Errorf("rpm: unable to seek: %w", err)
 	}
-	// Using an external tar (probably GNU tar) works where things like
-	// docker/docker's pkg/archive don't because it seems to set directory
-	// permissions later. Otherwise, some layers will have directories without
-	// the write bit created and the permissions set before their contents are
-	// created.
-	errbuf := bytes.Buffer{}
-	// Unprivileged containers can't call mknod(2), so exclude /dev and
-	// hopefully there aren't any others strewn about.
-	tarcmd := exec.CommandContext(ctx, "tar",
-		"--extract",
-		"--exclude", "dev",
-		"--exclude", ".wh*",
-		"--delay-directory-restore",
-		"--no-same-owner",
-		"--no-same-permissions",
+	tr = tar.NewReader(rd)
```

Comment:
This looks good, the actual home-rolled extraction seems like something that might be useful in more places, should we consider pulling it out into it's own function?